include/openmc/geometry.h

```diff
@@ -60,20 +60,61 @@ int cell_instance_at_level(const Particle& p, int level);
 //! \return True if the particle's location could be found and ascribed to a
 //!   valid geometry coordinate stack.
 //==============================================================================
-bool exhaustive_find_cell(Particle& p);
-bool neighbor_list_find_cell(Particle& p); // Only usable on surface crossings
+bool exhaustive_find_cell(Geometron& p, bool verbose = false);
+bool neighbor_list_find_cell(
+  Geometron& p, bool verbose = false); // Only usable on surface crossings
 
 //==============================================================================
 //! Move a particle into a new lattice tile.
 //==============================================================================
 
-void cross_lattice(Particle& p, const BoundaryInfo& boundary);
+void cross_lattice(
+  Geometron& p, const BoundaryInfo& boundary, bool verbose = false);
 
 //==============================================================================
 //! Find the next boundary a particle will intersect.
 //==============================================================================
 
-BoundaryInfo distance_to_boundary(Particle& p);
+BoundaryInfo distance_to_boundary(Geometron& p);
+
+/* Geometry routines can potentially throw this type of exception
+ */
+class ParticleLost : public std::exception {
```

Comment:
Ah yes, this makes perfect sense. Will refactor correspondingly.

Reply:
This is a bit different from the error handling with `fatal_error` vs. exceptions, because we want to continue running the program in this case but stop the particle.

Reply:
Changed to not use exceptions, yay!